### ISSUE TITLE
Internal Server Error when saving Marc Node containing Integer typed data field

### ISSUE THREAD

**Issue by shaunmuscat** [rism-ch/muscat#600]
I have created a new database table containing integer typed fields and configured a MARC node for this model within the Source MARC configuration.

When creating or editing a source, if data is added to these fields and the source is saved an internal server error occurs and the data is not saved. 

This internal error is caused by the marc node checking if the value is empty, since the .empty? method doesn't exist for the Fixnum class.

This is the stack trace.
```
D, [2018-02-13T11:18:35.298285 #85612] DEBUG -- : 
D, [2018-02-13T11:18:35.298375 #85612] DEBUG -- : 
I, [2018-02-13T11:18:35.298531 #85612]  INFO -- : Started POST "/admin/sources/marc_editor_save" for 127.0.0.1 at 2018-02-13 11:18:35 +1100
I, [2018-02-13T11:18:35.320275 #85612]  INFO -- : Processing by Admin::SourcesController#marc_editor_save as JSON
I, [2018-02-13T11:18:35.320559 #85612]  INFO -- :   Parameters: {"marc"=>"{\"fields\":[{\"001\":\"3\"},{\"031\":{\"subfields\":[{\"a\":\"test3\"},{\"b\":\"test3\"},{\"c\":\"test3\"},{\"o\":\"test3\"}]}},{\"040\":{\"subfields\":[{\"a\":\"AuArU\"},{\"b\":\"eng\"}]}},{\"240\":{\"subfields\":[{\"m\":\"test3\"},{\"0\":\"3\"}]}},{\"245\":{\"subfields\":[{\"a\":\"test3\"}]}},{\"650\":{\"subfields\":[{\"0\":\"3\"}]}},{\"852\":{\"subfields\":[{\"c\":\"test3\"},{\"e\":\"test3\"},{\"x\":\"6\"}]}},{\"999\":{\"subfields\":[{\"a\":\"1\"}]}}]}", "id"=>"3", "lock_version"=>"1", "record_type"=>"2", "record_status"=>"published", "record_owner"=>"2", "record_audit"=>"full", "triggers"=>"{}", "redirect"=>"false"}
D, [2018-02-13T11:18:35.325983 #85612] DEBUG -- :   User Load (2.2ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 2  ORDER BY `users`.`id` ASC LIMIT 1
D, [2018-02-13T11:18:35.330306 #85612] DEBUG -- :   StandardTitle Load (0.7ms)  SELECT  `standard_titles`.* FROM `standard_titles` WHERE `standard_titles`.`id` = 3 LIMIT 1
D, [2018-02-13T11:18:35.332061 #85612] DEBUG -- :   StandardTerm Load (0.4ms)  SELECT  `standard_terms`.* FROM `standard_terms` WHERE `standard_terms`.`id` = 3 LIMIT 1
D, [2018-02-13T11:18:35.333981 #85612] DEBUG -- :   Institution Load (0.6ms)  SELECT  `institutions`.* FROM `institutions` WHERE `institutions`.`id` = 6 LIMIT 1
D, [2018-02-13T11:18:35.336225 #85612] DEBUG -- :   CanonicTechnique Load (0.6ms)  SELECT  `canonic_techniques`.* FROM `canonic_techniques` WHERE `canonic_techniques`.`relation_numerator` = 1  ORDER BY `canonic_techniques`.`id` ASC LIMIT 1
I, [2018-02-13T11:18:35.341171 #85612]  INFO -- : Completed 500 Internal Server Error in 20ms (ActiveRecord: 4.4ms)
F, [2018-02-13T11:18:35.342732 #85612] FATAL -- : 
NoMethodError (undefined method `empty?' for 1:Fixnum):
  lib/marc_node.rb:91:in `block in resolve_externals'
  lib/marc_node.rb:70:in `each'
  lib/marc_node.rb:70:in `resolve_externals'
  lib/marc_node.rb:46:in `block in resolve_externals'
  lib/marc_node.rb:45:in `each'
  lib/marc_node.rb:45:in `resolve_externals'
  lib/marc.rb:202:in `load_from_hash'
  lib/marc_controller_actions.rb:31:in `block in included'


I, [2018-02-13T11:18:35.576164 #85612]  INFO -- :   Rendered /Users/shaun/.rvm/gems/ruby-2.3.0@muscat/gems/actionpack-4.2.10/lib/action_dispatch/middleware/templates/rescues/_source.erb (4.6ms)
I, [2018-02-13T11:18:35.601607 #85612]  INFO -- :   Rendered /Users/shaun.rvm/gems/ruby-2.3.0@muscat/gems/actionpack-4.2.10/lib/action_dispatch/middleware/templates/rescues/_trace.text.erb (1.3ms)
I, [2018-02-13T11:18:35.627936 #85612]  INFO -- :   Rendered /Users/shaun/.rvm/gems/ruby-2.3.0@muscat/gems/actionpack-4.2.10/lib/action_dispatch/middleware/templates/rescues/_request_and_response.text.erb (1.7ms)
I, [2018-02-13T11:18:35.628056 #85612]  INFO -- :   Rendered /Users/shaun/.rvm/gems/ruby-2.3.0@muscat/gems/actionpack-4.2.10/lib/action_dispatch/middleware/templates/rescues/diagnostics.text.erb (123.1ms)
```

**Comment by xhero** [comment]
Currently all the remote values should be strings (as they are cached directly from the MARC data which uses only strings), probably for "idiomatic" marc it could be better encode it  (as for relator codes)

**Comment by shaunmuscat** [comment]
For some of the data fields we intend restrict the user so that they can only submit numeric data as valid input. Using an integer typed database field achieves this for the authorities menu as active admin appears to restrict the field to numeric input only.

Since the MARC data only accepts strings, is there some existing way within Muscat to define the data field as string but restrict the user input to numeric data? I imagine it might be possible to hook some javascript into the Source and Authorities UI to achieve this, but is there anywhere within the existing MARC definitions where this is done?

**Comment by HirschSt** [comment]
Probably input validation? For now we have two different approaches with validation, one JS, the other ActiveRecord (serverside). If you want a strict user-input validation, you probably would like to go for the AR. For warnings I just have added an experimental branch with less strict rules, see https://github.com/rism-ch/muscat/tree/4.1-devel-warnings

**Comment by shaunmuscat** [comment]
@HirschSt I'm branched off v4.0.1 and cherry picked the commits from 4.1-devel-warnings into my branch. Added the following AR validations
```
  validates :relation_numerator, numericality: true
  validates :relation_denominator, numericality: true
  validates :temporal_offset, numericality: true
```
and SourceValidation warnings
```
"999":
  tags:
    a:
      warnings:
      - should_be_numeric
    b:
      warnings:
      - should_be_numeric
    e:
      warnings:
      - should_be_numeric
```

The warnings seem to work when creating the associated element but *not* when updating. For example, when Source doesn't have any of the "999" node associated and invalid data is inputted the new warning is displayed as expected. However when updating the data of an existing associated "999" node through a Source the validation fails and the rest of the Source is saved (as expected) but the association is dropped without any user notification. Here is the warning from console:
```
I, [2018-02-16T12:47:57.344128 #43821]  INFO -- : Started POST "/admin/sources/marc_editor_save" for 127.0.0.1 at 2018-02-16 12:47:57 +1100
I, [2018-02-16T12:47:57.354506 #43821]  INFO -- : Processing by Admin::SourcesController#marc_editor_save as JSON
I, [2018-02-16T12:47:57.354686 #43821]  INFO -- :   Parameters: {"marc"=>"{\"fields\":[{\"001\":\"6\"},{\"031\":{\"subfields\":[{\"a\":\"test3\"},{\"b\":\"test3\"},{\"c\":\"test3\"},{\"o\":\"test3\"}]}},{\"040\":{\"subfields\":[{\"a\":\"AuArU\"},{\"b\":\"eng\"}]}},{\"240\":{\"subfields\":[{\"m\":\"test3\"},{\"0\":\"3\"}]}},{\"245\":{\"subfields\":[{\"a\":\"test3\"}]}},{\"650\":{\"subfields\":[{\"0\":\"3\"}]}},{\"852\":{\"subfields\":[{\"c\":\"test3\"},{\"d\":\"test3\"},{\"e\":\"test3\"},{\"x\":\"6\"}]}},{\"999\":{\"subfields\":[{\"a\":\"5c\"},{\"b\":\"5\"},{\"c\":\"semibreve(s)\"},{\"d\":\"below\"},{\"e\":\"6\"},{\"f\":\"unison\"},{\"g\":\"7\"},{\"h\":\"r\"}]}}]}", "id"=>"6", "lock_version"=>"23", "record_type"=>"2", "record_status"=>"published", "record_owner"=>"2", "record_audit"=>"full", "triggers"=>"{}", "redirect"=>"false"}
D, [2018-02-16T12:47:57.356491 #43821] DEBUG -- :   User Load (0.5ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 2  ORDER BY `users`.`id` ASC LIMIT 1
D, [2018-02-16T12:47:57.358999 #43821] DEBUG -- :   StandardTitle Load (0.5ms)  SELECT  `standard_titles`.* FROM `standard_titles` WHERE `standard_titles`.`id` = 3 LIMIT 1
D, [2018-02-16T12:47:57.359831 #43821] DEBUG -- :   StandardTerm Load (0.3ms)  SELECT  `standard_terms`.* FROM `standard_terms` WHERE `standard_terms`.`id` = 3 LIMIT 1
D, [2018-02-16T12:47:57.361396 #43821] DEBUG -- :   Institution Load (0.5ms)  SELECT  `institutions`.* FROM `institutions` WHERE `institutions`.`id` = 6 LIMIT 1
D, [2018-02-16T12:47:57.365154 #43821] DEBUG -- :   CanonicTechnique Load (1.7ms)  SELECT  `canonic_techniques`.* FROM `canonic_techniques` WHERE `canonic_techniques`.`relation_numerator` = '5c' AND `canonic_techniques`.`relation_denominator` = '5' AND `canonic_techniques`.`interval` = 'semibreve(s)' AND `canonic_techniques`.`interval_direction` = 'below' AND `canonic_techniques`.`temporal_offset` = '6' AND `canonic_techniques`.`offset_units` = 'unison' AND `canonic_techniques`.`mensurations` = '7' AND `canonic_techniques`.`rubric` = 'r'  ORDER BY `canonic_techniques`.`id` ASC LIMIT 1
D, [2018-02-16T12:47:57.367079 #43821] DEBUG -- :    (0.4ms)  BEGIN
D, [2018-02-16T12:47:57.374508 #43821] DEBUG -- :    (0.3ms)  ROLLBACK
D, [2018-02-16T12:47:57.377586 #43821] DEBUG -- :   Source Load (0.5ms)  SELECT  `sources`.* FROM `sources` WHERE `sources`.`id` = 6 LIMIT 1

Marc Node Import error
Validation failed: Relation numerator is not a number
While importing: =999  ##$a5c$b5$csemibreve(s)$dbelow$e6$funison$g7$hr

D, [2018-02-16T12:47:57.382864 #43821] DEBUG -- :   Role Load (0.8ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_roles`.`role_id` WHERE `users_roles`.`user_id` = 2 AND (((roles.name = 'cataloger') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)))
D, [2018-02-16T12:47:57.385595 #43821] DEBUG -- :   Role Load (0.9ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_roles`.`role_id` WHERE `users_roles`.`user_id` = 2 AND (((roles.name = 'cataloger_prints') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)))
D, [2018-02-16T12:47:57.388581 #43821] DEBUG -- :   Role Load (1.1ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_roles`.`role_id` WHERE `users_roles`.`user_id` = 2 AND (((roles.name = 'editor') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)))
D, [2018-02-16T12:47:57.391388 #43821] DEBUG -- :   Role Load (0.8ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_roles`.`role_id` WHERE `users_roles`.`user_id` = 2 AND (((roles.name = 'admin') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)))
D, [2018-02-16T12:47:57.392879 #43821] DEBUG -- :   CACHE (0.0ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_roles`.`role_id` WHERE `users_roles`.`user_id` = 2 AND (((roles.name = 'editor') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)))  [["user_id", 2]]
D, [2018-02-16T12:47:57.393547 #43821] DEBUG -- :   CACHE (0.0ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_roles`.`role_id` WHERE `users_roles`.`user_id` = 2 AND (((roles.name = 'admin') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)))  [["user_id", 2]]
D, [2018-02-16T12:47:57.394298 #43821] DEBUG -- :   User Load (0.4ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 2 LIMIT 1
D, [2018-02-16T12:47:57.395760 #43821] DEBUG -- :   CACHE (0.0ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_roles`.`role_id` WHERE `users_roles`.`user_id` = 2 AND (((roles.name = 'cataloger') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)))  [["user_id", 2]]
D, [2018-02-16T12:47:57.396755 #43821] DEBUG -- :   CACHE (0.0ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_roles`.`role_id` WHERE `users_roles`.`user_id` = 2 AND (((roles.name = 'editor') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)))  [["user_id", 2]]
D, [2018-02-16T12:47:57.397562 #43821] DEBUG -- :   CACHE (0.0ms)  SELECT `roles`.* FROM `roles` INNER JOIN `users_roles` ON `roles`.`id` = `users_roles`.`role_id` WHERE `users_roles`.`user_id` = 2 AND (((roles.name = 'admin') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)))  [["user_id", 2]]
D, [2018-02-16T12:47:57.398706 #43821] DEBUG -- :    (0.4ms)  BEGIN
D, [2018-02-16T12:47:57.400917 #43821] DEBUG -- :   Source Load (0.7ms)  SELECT  `sources`.* FROM `sources` WHERE `sources`.`id` = 6 LIMIT 1
D, [2018-02-16T12:47:57.408689 #43821] DEBUG -- :   StandardTitle Load (0.6ms)  SELECT  `standard_titles`.* FROM `standard_titles` WHERE `standard_titles`.`id` = 3 LIMIT 1
D, [2018-02-16T12:47:57.409839 #43821] DEBUG -- :   StandardTerm Load (0.4ms)  SELECT  `standard_terms`.* FROM `standard_terms` WHERE `standard_terms`.`id` = 3 LIMIT 1
D, [2018-02-16T12:47:57.410630 #43821] DEBUG -- :   Institution Load (0.3ms)  SELECT  `institutions`.* FROM `institutions` WHERE `institutions`.`id` = 6 LIMIT 1
D, [2018-02-16T12:47:57.412577 #43821] DEBUG -- :   CanonicTechnique Load (0.8ms)  SELECT  `canonic_techniques`.* FROM `canonic_techniques` WHERE `canonic_techniques`.`id` = 32 LIMIT 1
D, [2018-02-16T12:47:57.417301 #43821] DEBUG -- :   SQL (0.5ms)  UPDATE `sources` SET `sources`.`marc_source` = '=001  6\r\n=031  ##$atest3$btest3$ctest3$otest3\r\n=040  ##$aAuArU$beng\r\n=240  10$atest3$mtest3$03\r\n=245  10$atest3\r\n=650  07$atest3$03\r\n=852  ##$atest3$ctest3$dtest3$etest3$x6\r\n=999  ##$a5c$b5$csemibreve(s)$dbelow$e6$funison$g7$hr\r\n', `sources`.`updated_at` = '2018-02-16 01:47:57', `sources`.`lock_version` = 24 WHERE `sources`.`id` = 6 AND `sources`.`lock_version` = 23
D, [2018-02-16T12:47:57.419147 #43821] DEBUG -- :   PaperTrail::Version Load (0.5ms)  SELECT  `versions`.* FROM `versions` WHERE `versions`.`item_id` = 6 AND `versions`.`item_type` = 'Source'  ORDER BY `versions`.`created_at` DESC, `versions`.`id` DESC LIMIT 1
D, [2018-02-16T12:47:57.420339 #43821] DEBUG -- :   CACHE (0.0ms)  SELECT  `versions`.* FROM `versions` WHERE `versions`.`item_id` = 6 AND `versions`.`item_type` = 'Source'  ORDER BY `versions`.`created_at` DESC, `versions`.`id` DESC LIMIT 1  [["item_id", 6], ["item_type", "Source"]]
D, [2018-02-16T12:47:57.421040 #43821] DEBUG -- :   CACHE (0.0ms)  SELECT  `versions`.* FROM `versions` WHERE `versions`.`item_id` = 6 AND `versions`.`item_type` = 'Source'  ORDER BY `versions`.`created_at` DESC, `versions`.`id` DESC LIMIT 1  [["item_id", 6], ["item_type", "Source"]]
D, [2018-02-16T12:47:57.421975 #43821] DEBUG -- :   CACHE (0.0ms)  SELECT  `versions`.* FROM `versions` WHERE `versions`.`item_id` = 6 AND `versions`.`item_type` = 'Source'  ORDER BY `versions`.`created_at` DESC, `versions`.`id` DESC LIMIT 1  [["item_id", 6], ["item_type", "Source"]]
D, [2018-02-16T12:47:57.422703 #43821] DEBUG -- :   CACHE (0.0ms)  SELECT  `versions`.* FROM `versions` WHERE `versions`.`item_id` = 6 AND `versions`.`item_type` = 'Source'  ORDER BY `versions`.`created_at` DESC, `versions`.`id` DESC LIMIT 1  [["item_id", 6], ["item_type", "Source"]]
D, [2018-02-16T12:47:57.427365 #43821] DEBUG -- :   Person Load (0.6ms)  SELECT `people`.* FROM `people` INNER JOIN `sources_to_people` ON `people`.`id` = `sources_to_people`.`person_id` WHERE `sources_to_people`.`source_id` = 6
D, [2018-02-16T12:47:57.429837 #43821] DEBUG -- :   StandardTitle Load (0.8ms)  SELECT `standard_titles`.* FROM `standard_titles` INNER JOIN `sources_to_standard_titles` ON `standard_titles`.`id` = `sources_to_standard_titles`.`standard_title_id` WHERE `sources_to_standard_titles`.`source_id` = 6
D, [2018-02-16T12:47:57.432732 #43821] DEBUG -- :   StandardTerm Load (0.7ms)  SELECT `standard_terms`.* FROM `standard_terms` INNER JOIN `sources_to_standard_terms` ON `standard_terms`.`id` = `sources_to_standard_terms`.`standard_term_id` WHERE `sources_to_standard_terms`.`source_id` = 6
D, [2018-02-16T12:47:57.435462 #43821] DEBUG -- :   Place Load (0.6ms)  SELECT `places`.* FROM `places` INNER JOIN `sources_to_places` ON `places`.`id` = `sources_to_places`.`place_id` WHERE `sources_to_places`.`source_id` = 6
D, [2018-02-16T12:47:57.437842 #43821] DEBUG -- :   LiturgicalFeast Load (0.5ms)  SELECT `liturgical_feasts`.* FROM `liturgical_feasts` INNER JOIN `sources_to_liturgical_feasts` ON `liturgical_feasts`.`id` = `sources_to_liturgical_feasts`.`liturgical_feast_id` WHERE `sources_to_liturgical_feasts`.`source_id` = 6
D, [2018-02-16T12:47:57.439644 #43821] DEBUG -- :   Catalogue Load (0.5ms)  SELECT `catalogues`.* FROM `catalogues` INNER JOIN `sources_to_catalogues` ON `catalogues`.`id` = `sources_to_catalogues`.`catalogue_id` WHERE `sources_to_catalogues`.`source_id` = 6
D, [2018-02-16T12:47:57.441263 #43821] DEBUG -- :   Institution Load (0.5ms)  SELECT `institutions`.* FROM `institutions` INNER JOIN `sources_to_institutions` ON `institutions`.`id` = `sources_to_institutions`.`institution_id` WHERE `sources_to_institutions`.`source_id` = 6
D, [2018-02-16T12:47:57.443385 #43821] DEBUG -- :   Source Load (0.5ms)  SELECT `sources`.* FROM `sources` INNER JOIN `sources_to_sources` ON `sources`.`id` = `sources_to_sources`.`source_b_id` WHERE `sources_to_sources`.`source_a_id` = 6
D, [2018-02-16T12:47:57.444760 #43821] DEBUG -- :   Work Load (0.4ms)  SELECT `works`.* FROM `works` INNER JOIN `sources_to_works` ON `works`.`id` = `sources_to_works`.`work_id` WHERE `sources_to_works`.`source_id` = 6
D, [2018-02-16T12:47:57.446307 #43821] DEBUG -- :   CanonicTechnique Load (0.6ms)  SELECT `canonic_techniques`.* FROM `canonic_techniques` INNER JOIN `sources_to_canonic_techniques` ON `canonic_techniques`.`id` = `sources_to_canonic_techniques`.`canonic_technique_id` WHERE `sources_to_canonic_techniques`.`source_id` = 6

Foreign Links: Could not add a record () in the relationship with 6 (Source)
- Added records dump: [#<CanonicTechnique id: nil, relation_numerator: "5c", relation_denominator: "5", interval: "semibreve(s)", interval_direction: "below", temporal_offset: "6", offset_units: "unison", mensurations: "7", rubric: "r", canon_type_id: nil, wf_audit: 0, wf_stage: 1, wf_notes: nil, wf_owner: 2, wf_version: 0, created_at: nil, updated_at: nil, lock_version: 0>]
Validation failed: Relation numerator is not a number
D, [2018-02-16T12:47:57.448770 #43821] DEBUG -- :   SQL (0.8ms)  DELETE FROM `sources_to_canonic_techniques` WHERE `sources_to_canonic_techniques`.`source_id` = 6 AND `sources_to_canonic_techniques`.`canonic_technique_id` = 32
D, [2018-02-16T12:47:57.453780 #43821] DEBUG -- :   SQL (0.5ms)  INSERT INTO `delayed_jobs` (`handler`, `queue`, `parent_id`, `parent_type`, `run_at`, `created_at`, `updated_at`) VALUES ('--- !ruby/object:ReindexForeignRelationsJob\nparent_id: 6\nrelations:\n- :class: &1 !ruby/class \'CanonicTechnique\'\n  :id: \n- :class: *1\n  :id: 32\n', 'authority', 6, 'Source', '2018-02-16 01:47:57', '2018-02-16 01:47:57', '2018-02-16 01:47:57')
D, [2018-02-16T12:47:57.467350 #43821] DEBUG -- :   SOLR Request (6.6ms)  [ path=update parameters={} ]
D, [2018-02-16T12:47:57.468445 #43821] DEBUG -- :    (0.5ms)  COMMIT
D, [2018-02-16T12:47:57.469548 #43821] DEBUG -- :   Source Load (0.5ms)  SELECT  `sources`.* FROM `sources` WHERE `sources`.`id` = 6 LIMIT 1
D, [2018-02-16T12:47:57.476093 #43821] DEBUG -- :   StandardTitle Load (0.5ms)  SELECT  `standard_titles`.* FROM `standard_titles` WHERE `standard_titles`.`id` = 3 LIMIT 1
D, [2018-02-16T12:47:57.477155 #43821] DEBUG -- :   StandardTerm Load (0.3ms)  SELECT  `standard_terms`.* FROM `standard_terms` WHERE `standard_terms`.`id` = 3 LIMIT 1
D, [2018-02-16T12:47:57.477937 #43821] DEBUG -- :   Institution Load (0.4ms)  SELECT  `institutions`.* FROM `institutions` WHERE `institutions`.`id` = 6 LIMIT 1
Tag 999: missing master (expected in $0), tag contents: =999  ##$a5c$b5$csemibreve(s)$dbelow$e6$funison$g7$hr
 
D, [2018-02-16T12:47:57.512420 #43821] DEBUG -- :   SOLR Request (32.7ms)  [ path=update parameters={} ]
I, [2018-02-16T12:47:57.512954 #43821]  INFO -- : Completed 200 OK in 158ms (Views: 0.1ms | ActiveRecord: 22.3ms | Solr: 6.6ms)
D, [2018-02-16T12:47:57.528281 #43821] DEBUG -- : 
D, [2018-02-16T12:47:57.528390 #43821] DEBUG -- : 
I, [2018-02-16T12:47:57.528498 #43821]  INFO -- : Started GET "/admin/sources/6/edit" for 127.0.0.1 at 2018-02-16 12:47:57 +1100
I, [2018-02-16T12:47:57.539800 #43821]  INFO -- : Processing by Admin::SourcesController#edit as HTML
I, [2018-02-16T12:47:57.539882 #43821]  INFO -- :   Parameters: {"id"=>"6"}
D, [2018-02-16T12:47:57.541860 #43821] DEBUG -- :   User Load (0.5ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 2  ORDER BY `users`.`id` ASC LIMIT 1
D, [2018-02-16T12:47:57.543149 #43821] DEBUG -- :   Source Load (0.5ms)  SELECT  `sources`.* FROM `sources` WHERE `sources`.`id` = 6 LIMIT 1
D, [2018-02-16T12:47:57.545662 #43821] DEBUG -- :   StandardTitle Load (0.6ms)  SELECT  `standard_titles`.* FROM `standard_titles` WHERE `standard_titles`.`id` = 3 LIMIT 1
Tag 999: missing master (expected in $0), tag contents: =999  ##$a5c$b5$csemibreve(s)$dbelow$e6$funison$g7$hr
```

However this is the kind of thing I'd be after, so I'll try to extend and contribute where feasible.

**Comment by HirschSt** [comment]
Thanks, this behaviour should be resolved with d4ea5232587f5461fd4729648e032b441e290965
